packages/plugin-birdeye/src/providers/defi/price-history-provider.ts

```diff
@@ -0,0 +1,230 @@
+import {
+    IAgentRuntime,
+    Memory,
+    Provider,
+    State,
+    elizaLogger,
+} from "@elizaos/core";
+import {
+    BASE_URL,
+    Chain,
+    extractChain,
+    extractContractAddresses,
+    extractTimeRange,
+    formatTimestamp,
+    formatValue,
+    makeApiRequest,
+} from "../utils";
+
+// Types
+interface PriceHistoryData {
+    price: number;
+    timestamp: number;
+    volume?: number;
+}
+
+interface PriceHistoryResponse {
+    data: PriceHistoryData[];
+    token: string;
+}
+
+// Constants
+const PRICE_HISTORY_KEYWORDS = [
+    "price history",
+    "historical price",
+    "price chart",
+    "price trend",
+    "price movement",
+    "price changes",
+    "price over time",
+    "price timeline",
+    "price performance",
+    "price data",
+    "historical data",
+    "price analysis",
+    "price tracking",
+    "price evolution",
+] as const;
+
+// Helper functions
+const containsPriceHistoryKeyword = (text: string): boolean => {
+    return PRICE_HISTORY_KEYWORDS.some((keyword) =>
+        text.toLowerCase().includes(keyword.toLowerCase())
+    );
+};
+
+const getPriceHistory = async (
+    apiKey: string,
+    contractAddress: string,
+    startTime: number,
+    endTime: number,
+    chain: Chain
+): Promise<PriceHistoryResponse | null> => {
+    try {
+        const params = new URLSearchParams({
+            address: contractAddress,
+            time_from: startTime.toString(),
+            time_to: endTime.toString(),
+        });
+        const url = `${BASE_URL}/defi/price_history_unix?${params.toString()}`;
```

Comment:
```
 ℹ INFORMATIONS
   Fetching price history for token 0x0b3e328455c4059eeb9e3f84b5543f74e24e7e1b from 11/27/2024, 8:08:14 PM to 12/27/2024, 8:08:14 PM on base with 1 day interval from: 
   https://public-api.birdeye.so/defi/price_history_unix?address=0x0b3e328455c4059eeb9e3f84b5543f74e24e7e1b&time_from=1732766894&time_to=1735358894&interval=1d 

 ⛔ ERRORS
   API Error (404): 
   Resource not found 

 ⛔ ERRORS
   Error fetching price history: 
   Resource not found 
   ```
   
   The [correct endpoint ](https://docs.birdeye.so/reference/get_defi-historical-price-unix)should be `historical_price_unix`

Reply:
Actually this endpoint doesn't seem to support the evm chains yet.
```
{
  "success": false,
  "message": "Chain ethereum is not supported yet"
}
```